relay.go

```diff
@@ -694,3 +778,82 @@ func validateRelayMaxTimeout(d time.Duration, logger Logger) time.Duration {
 	).Warn("Configured RelayMaxTimeout is invalid, using default instead.")
 	return _defaultRelayMaxTimeout
 }
+
+type relayFragmentSender struct {
+	callReq            *lazyCallReq
+	framePool          FramePool
+	frameReceiver      frameReceiver
+	failRelayItemFunc  func(items *relayItems, id uint32, failure string)
+	outboundRelayItems *relayItems
+	origID             uint32
+}
+
+func (r *Relayer) newFragmentSender(dstRelay frameReceiver, cr *lazyCallReq, origID uint32) *relayFragmentSender {
+	return &relayFragmentSender{
+		callReq:            cr,
+		framePool:          r.conn.opts.FramePool,
+		frameReceiver:      dstRelay,
+		failRelayItemFunc:  r.failRelayItem,
+		outboundRelayItems: r.outbound,
+		origID:             origID,
+	}
+}
+
+func (rfs *relayFragmentSender) newFragment(initial bool, checksum Checksum) (*writableFragment, error) {
+	frame := rfs.framePool.Get()
+	frame.Header.ID = rfs.callReq.Header.ID
+	if initial {
+		frame.Header.messageType = messageTypeCallReq
+	} else {
+		frame.Header.messageType = messageTypeCallReqContinue
+	}
+
+	contents := typed.NewWriteBuffer(frame.Payload[:])
+
+	// flags:1
+	flagsRef := contents.DeferByte()
+	flagsRef.Update(rfs.callReq.Payload[_flagsIndex])
+
+	if initial {
+		// Copy all data before the checksum for the initial frame
+		contents.WriteBytes(rfs.callReq.Payload[_flagsIndex+1 : rfs.callReq.checksumTypeOffset])
+	}
+
+	// checksumType:1
+	contents.WriteSingleByte(byte(checksum.TypeCode()))
+
+	// checksum: checksum.Size()
+	checksumRef := contents.DeferBytes(checksum.Size())
+
+	if initial {
+		// arg1~1: write arg1 to the initial frame
+		contents.WriteUint16(uint16(len(rfs.callReq.method)))
+		contents.WriteBytes(rfs.callReq.method)
+		checksum.Add(rfs.callReq.method)
+	}
+
+	return &writableFragment{
+		flagsRef:    flagsRef,
+		checksumRef: checksumRef,
+		checksum:    checksum,
+		contents:    contents,
+		frame:       frame,
+	}, contents.Err()
+}
+
+func (rfs *relayFragmentSender) flushFragment(f *writableFragment) error {
+	frame, ok := f.frame.(*Frame)
+	if !ok {
+		return fmt.Errorf("got unexpected frame type: %t", f.frame)
```

Comment:
should be `%T` for type.

however, this whole thing is unnecessary, not sure why we had a `interface{}`, cleaned up:
https://github.com/uber/tchannel-go/pull/804